### PR TITLE
Lh/search button toggle

### PR DESCRIPTION
**Pull request by lucyhutcheson**
**Related Issue(s):**

- N/A

**Proposed Changes:**

1.  Add a toggle switch to enable auto-search or manual search

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/Element84/filmdrop-ui/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: src/components/Search/Search.js [hunk 1]
```diff
@@ -547,7 +566,12 @@ const Search = () => {
     }
 
     if (showCloudSliderRef.current) {
-      createMosaicBody.query = getCloudCoverQueryVal(_cloudCover)
+      createMosaicBody.query = {
```
Comment (philvarner) [inline src/components/Search/Search.js:569]: ideally, this would still be a shared function between mosaic and scene -- can that be done?

Comment (lucyhutcheson) [reply src/components/Search/Search.js:569]: Added the function in.

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -557,6 +578,7 @@ const Search = () => {
       },
       body: JSON.stringify(createMosaicBody)
     }
+    console.log(requestOptions)
```
Comment (bradleyandrick) [inline src/components/Search/Search.js:581]: don't know if we need this log here? I tend to keep logs out of main unless required

File: src/utils/debounce.js [hunk 3]
```diff
@@ -1,12 +1,22 @@
 // throttle function to prevent map from rendering too quickly
-export default function debounce(func, waitInMillis) {
+export default function debounce(func, wait, immediate) {
```
Comment (bradleyandrick) [inline src/utils/debounce.js:2]: I don't see the third 'immediate' param being passed anywhere, is that needed?

File: src/components/Search/SearchAPI.js [hunk 4]
```diff
@@ -52,5 +52,10 @@ export const fetchAggregatedItems = async (
     })
     .filter((x) => x)
 
-  return { type: 'AggregatedResults', features: mappedKeys, numberMatched }
+  return {
+    type: 'FeatureCollection',
+    features: mappedKeys,
+    numberMatched,
+    searchType: 'AggregatedResults'
+  }
```
Comment (bradleyandrick) [inline src/components/Search/SearchAPI.js:60]: _gonna post this same comment a few places to show where I was seeing it:_

there are some changes in here that don't seem to be related to adding in the search button toggle. I know another comment said that this was branched lh/aggregate-results, did some of that make it into here?

File: src/components/Search/Search.js [hunk 5]
```diff
@@ -404,14 +421,16 @@ const Search = () => {
...
+      if (response) {
+        dispatch(setSearchResults(response))
+        searchResultsRef.current = response
+        dispatch(setSearchLoading(false))
+
+        // add new footprints to map
+        const resultFootprintsFound = L.geoJSON(response, options)
+        resultFootprintsFound.id = 'resultLayer'
+        resultFootprintsFound.addTo(resultFootprintsRef.current)
+      }
```
Comment (bradleyandrick) [inline src/components/Search/Search.js:433]: _gonna post this same comment a few places to show where I was seeing it:_

there are some changes in here that don't seem to be related to adding in the search button toggle. I know another comment said that this was branched lh/aggregate-results, did some of that make it into here?

File: src/components/Search/Search.js [hunk 6]
```diff
@@ -375,8 +392,8 @@ const Search = () => {
       (zoomLevelRef.current <= 8 && selectedCollectionRef.current === 'naip')
     ) {
       // LOW ZOOM - HEATMAP HEXGRID VIEW
-      typeOfSearch = { type: 'scene' }
-      searchTypeRef.current = 'scene'
+      typeOfSearch = { type: 'aggregated' }
+      searchTypeRef.current = 'aggregated'
```
Comment (bradleyandrick) [inline src/components/Search/Search.js:396]: _gonna post this same comment a few places to show where I was seeing it:_

there are some changes in here that don't seem to be related to adding in the search button toggle. I know another comment said that this was branched lh/aggregate-results, did some of that make it into here?

File: src/components/Layout/Content/BottomContent/BottomContent.js [hunk 7]
```diff
@@ -76,13 +76,13 @@ const BottomContent = () => {
...
       {_searchResults?.numberMatched &&
-      _searchResults?.type !== 'AggregatedResults' ? (
+      _searchResults?.searchType !== 'AggregatedResults' ? (
         <div className="resultCount">
           Showing {_searchResults.numberReturned} of{' '}
           {_searchResults.numberMatched} Scenes
         </div>
       ) : null}
-      {_searchResults?.type === 'AggregatedResults' ? (
+      {_searchResults?.searchType === 'AggregatedResults' ? (
```
Comment (bradleyandrick) [inline src/components/Layout/Content/BottomContent/BottomContent.js:85]: _gonna post this same comment a few places to show where I was seeing it:_

there are some changes in here that don't seem to be related to adding in the search button toggle. I know another comment said that this was branched lh/aggregate-results, did some of that make it into here?

**Review by bradleyandrick — COMMENTED** [review]
a few comments

File: src/components/Search/Search.js [hunk 8]
```diff
@@ -641,6 +663,21 @@ const Search = () => {
           <ViewSelector></ViewSelector>
         </div>
       )}
+      <div className="searchContainer searchButton">
+        {!autoSearchSwitch && (
+          <button className="actionButton" onClick={() => processSearchBtn()}>
+            Search
+          </button>
+        )}
```
[the remark this answers is not in the document]
Comment (lucyhutcheson) [reply src/components/Search/Search.js:671]: Added styling to disable the button and leave it in place.

File: src/components/Search/Search.js [hunk 9]
```diff
@@ -641,6 +663,21 @@ const Search = () => {
...
+        <div className="autoSearchContainer">
+          <label>Auto Search</label>
+          <Switch
+            checked={autoSearchSwitch}
+            onChange={handleSwitchChange}
+            inputProps={{ 'aria-label': 'controlled' }}
+          />
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (lucyhutcheson) [reply src/components/Search/Search.js:680]: Added responsive styling along with shrinking the toggle to fit the space better. The toggle will wrap underneath the Search button on smaller screens.

File: src/components/Search/Search.js (hunk 2 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:581]: This has been removed.

File: src/components/Search/SearchAPI.js (hunk 4 above)
Comment (lucyhutcheson) [reply src/components/Search/SearchAPI.js:60]: My apologies for the confusion. This branch was a fork off the aggregated results branch and these commits were mistakenly included.

File: src/components/Search/Search.js (hunk 5 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:433]: My apologies for the confusion. This branch was a fork off the aggregated results branch and these commits were mistakenly included.

File: src/components/Layout/Content/BottomContent/BottomContent.js (hunk 7 above)
Comment (lucyhutcheson) [reply src/components/Layout/Content/BottomContent/BottomContent.js:85]: My apologies for the confusion. This branch was a fork off the aggregated results branch and these commits were mistakenly included.

File: src/components/Search/Search.js (hunk 6 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:396]: My apologies for the confusion. This branch was a fork off the aggregated results branch and these commits were mistakenly included.

File: src/utils/debounce.js (hunk 3 above)
Comment (lucyhutcheson) [reply src/utils/debounce.js:2]: This parameter has been added to the Search Button function.

**Comment by bradleyandrick** [comment]
approved 👍

**Review by bradleyandrick — APPROVED** [review]
